joss/paper.md

```diff
@@ -0,0 +1,151 @@
+On the educational side, the package includes visualization logic that allows students to
+interact with these structures in an exploratory manner, and has already seen usage in
+undergraduate courses. `automata` has already been cited in publications [@Erickson23], with more
+to come as the package becomes more popular.
+
+`automata` has seen a large number of contributions by external contributors and wide adoption,
+demonstrating the demand for a high-quality Python package providing these features. The code is
+well maintained, including a comprehensive test suite and type annotations, meaning new features
+can be incorporated from requests by the community at a rapid pace.
+
+# Example usage
+
+![A visualization of `target_words_dfa`. Transitions on characters leading to immediate rejections are omitted.\label{fig:target_words_dfa}](finite_language_dfa.png){ width=100% }
+
+The following example is inspired by the use case described in [@Johnson_2010].
+We wish to determine which strings in a given set are within the target edit distance
+to a reference string. We will do this by using utilities provided by `automata`,
+starting by initializing a DFA corresponding to the input set.
+
+```python
+from automata.fa.dfa import DFA
+from automata.fa.nfa import NFA
+import string
+
+input_symbols = set(string.ascii_lowercase)
+
+target_words = {'these', 'are', 'target', 'words', 'them', 'those'}
+
+target_words_dfa = DFA.from_finite_language(
+  input_symbols,
+  target_words,
+)
+```
+A visualization of `target_words_dfa`, generated by the package in a Jupyter notebook,
+is depicted in \autoref{fig:target_words_dfa}.
+
+Next, we construct an NFA recognizing all strings within the given edit distance of a
+reference string. This construction can again be done with functions provided by the library.
+We need to perform an NFA to DFA conversion for later.
+
+```python
+reference_string = 'they'
+edit_distance = 2
+
+words_within_edit_distance_dfa = DFA.from_nfa(
+  NFA.edit_distance(
+    input_symbols,
+    reference_string,
+    edit_distance,
+  )
+)
+```
+
+Finally, we take the intersection of the two DFAs we have constructed and read all of
+the words in the result into a list. The library makes this straightforward and idiomatic.
+
+```python
+found_words_dfa = target_words_dfa & words_within_edit_distance_dfa
+found_words = list(found_words_dfa)
+```
+
+# Acknowledgements
+
+Thanks (in no particular order) to GitHub users
+[YtvwlD](https://github.com/YtvwlD),
+[dengl11](https://github.com/dengl11),
+[Tagl](https://github.com/Tagl),
+[lewiuberg](https://github.com/lewiuberg),
+[CamiloMartinezM](https://github.com/CamiloMartinezM),
+[abhinavsinha‑adrino](https://github.com/abhinavsinha-adrino),
+[EduardoGoulart1](https://github.com/EduardoGoulart1), and
+[khoda81](https://github.com/khoda81)
+for their invaluable code contributions to this project.
+
+# References
```

Comment:
@eliotwrobson Is it fine to leave the _References_ section blank?

Reply:
This is automatically populated during the build.